client/client.go

```diff
@@ -0,0 +1,32 @@
+package client
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/websocket"
+	"github.com/taskcluster/webhooktunnel/wsmux"
+)
+
+// Client is used to connect to a proxy and serve endpoints
+// defined in Handler
+type Client struct {
+	Id string
+	// handler should be a mux to handle different end points
+	Handler http.Handler
+	Config  wsmux.Config
+}
+
+// ServeOnProxy serves endpoints registered on handler on the proxy
+func (c *Client) ServeOnProxy(proxyAddr string) error {
+	addr := strings.TrimSuffix(proxyAddr, "/") + "/register/" + c.Id + "/"
+
+	conn, _, err := websocket.DefaultDialer.Dial(addr, nil)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Insert a `// TODO: implement reconnect logic`, we don't have to do it just yet...